.github/workflows/shippable_builds.yml

```diff
@@ -224,21 +224,45 @@ jobs:
 
           ./gradlew :app-${APP_NAME}:printVersionInfo -PbuildType=${RELEASE_TYPE} -PflavorName=${PACKAGE_FLAVOR} --configure-on-demand
 
-      - name: Bump version code
-        id: bump_version_code
-        if: ${{ contains(matrix.releaseTarget, 'github') }}
+      - name: Determine new version code
+        id: new_version_code
+        if: ${{ contains(matrix.releaseTarget, 'github') || needs.dump_config.outputs.releaseType == 'daily' }}
         shell: bash
         env:
           APP_NAME: ${{ matrix.appName }}
           OLD_VERSION_CODE: ${{ steps.appinfo.outputs.VERSION_CODE }}
+          RELEASE_TYPE: ${{ vars.RELEASE_TYPE }}
         run: |
-          NEW_VERSION_CODE=$(($OLD_VERSION_CODE + 1))
+          if [[ "$RELEASE_TYPE" = "beta" || "$RELEASE_TYPE" = "release" ]]; then
+            NEW_VERSION_CODE=$(($OLD_VERSION_CODE + 1))
+          elif [[ "$RELEASE_TYPE" = "daily" ]]; then
+            VERSION_FILE="$(pwd)/version.properties"
+            if [[ -e "$VERSION_FILE" ]]; then
+              while IFS="=" read -r key value; do
+                if [[ "${key}" = "NEW_VERSION_CODE" ]]; then
+                  NEW_VERSION_CODE="${value}"
+                fi
+              done < $VERSION_FILE
+            else
+              # Daily uses date-based calculation for version code in yDDDHHmm format.
+              # y: Years since 2023 (e.g. 2024 is 1)
+              # DDD: Day of the year (3 digits, zero-padded)
+              # HH: Hour of the day (2 digits, 00-23)
+              # mm: Minute of the hour (2 digits)
+              year=$(( $(date +"%Y") - 2023 ))
+              day=$(printf "%03d" $(date +"%j"))
+              time=$(date +"%H%M")
+              NEW_VERSION_CODE="${year}${day}${time}"
```

Comment:
Can you add a safety check here so that the final number is not higher than what we'd expect at the end of 2025? I want to avoid a future error accidentally burning the version code, if we reach 2100000000 then we are cooked and have to throw away the package id. Same check with beta/release would make sense, assuming monthly releases and weekly betas.

Reply:
That's fair, let's make it a month earlier.